docs/platforms/Swift-Package-Manager.md

```diff
@@ -0,0 +1,645 @@
+
+9. Move all remaining files from `ios/Classes` to `Sources/plugin_name_ios`
+10. `ios/Assets`, `ios/Resources`, `ios/Classes` should now be empty and can be deleted
+11. If your header files were previously within the same directory as your implementation files, you may need to change your import statements.
+
+   For example, if the following changes were made:
+     * `ios/Classes/PublicHeaderFile.h` --> `Sources/plugin_name_ios/include/plugin_name_ios/PublicHeaderFile.h`
+     * `ios/Classes/ImplementationFile.m` --> `Sources/plugin_name_ios/ImplementationFile.m`
+
+   Within `ImplementationFile.m`, the import would change:
+   ```diff
+   - #import "PublicHeaderFile.h"
+   + #import "./include/plugin_name_ios/PublicHeaderFile.h"
+   ```
+
+12. If using pigeon, you'll want to update your pigeon input file
+```diff
+- objcHeaderOut: 'ios/Classes/messages.g.h',
++ objcHeaderOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.h',
+- objcSourceOut: 'ios/Classes/messages.g.m',
++ objcSourceOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.m',
+```
+
+If your `objcHeaderOut` file is no longer within the same directory as the `objcSourceOut`, you can change the `#import` using `ObjcOptions.headerIncludePath`:
+```diff
+objcHeaderOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/include/plugin_name_ios/messages.g.h',
+objcSourceOut: 'ios/plugin_name_ios/Sources/plugin_name_ios/messages.g.m',
++ objcOptions: ObjcOptions(
++   headerIncludePath: './include/plugin_name_ios/messages.g.h',
++ ),
+```
+
+13. Update your Package.swift with any customizations you may need
+    1. Open `/plugin_name/plugin_name_ios/ios/plugin_name_ios/` in Xcode
+        * If package does not show any files in Xcode, quit Xcode (Xcode > Quit Xcode) and reopen
+        * You don't need to edit your Package.swift through Xcode, but Xcode will provide helpful feedback
+        * If Xcode isn't updating after you make a change, try clicking File > Packages > Reset Package Caches
+    2. [Add dependencies](https://developer.apple.com/documentation/packagedescription/package/dependency)
+    3. If your package must be linked explicitly `static` or `dynamic` ([not recommended](https://developer.apple.com/documentation/packagedescription/product/library(name:type:targets:))), update the [Product](https://developer.apple.com/documentation/packagedescription/product) to define the type
+    ```swift
+    products: [
+        .library(name: "plugin-name-ios", type: .static, targets: ["plugin_name_ios"])
+    ],
+    ```
+    4. Make any other customizations - see https://developer.apple.com/documentation/packagedescription for more info on how to write a Package.swift.
+    5. If you add additional targets to your Package.swift, try to name them uniquely. If your target name conflicts with another target from another package, this can cause issues that may require manual intervention to be able to use your plugin.
```

Comment:
FYI, I did add this note about name collisions, but could be good to have a section on how to handle them too as you mentioned